docs/uds-package-practices.md

```diff
@@ -0,0 +1,67 @@
+# UDS Package Practices
+
+This document describes the practices that a UDS package **must**, **should**, and **may** follow to be considered officially supported as a UDS package and to be able to display the "Made for UDS" badge on its GitHub repository README.md file:
+
+[<img alt="Made for UDS" src="./made-for-uds.svg" height="20px"/>](https://github.com/defenseunicorns/uds-core)
+
+> [!NOTE]
+> This badge should link to the `uds-core` repo and should match the height of any other badges (i.e. `20px`).
+
+## Integrations
+
+Below are the minimal services that a UDS package **must** integrate with, using the [UDS `Package` custom resource](https://github.com/defenseunicorns/uds-core/blob/main/src/pepr/operator/README.md#example-uds-package-cr):
+
+### Istio
+
+- **Must** define any external interfaces under the `expose` key.
+- **Must** deploy and operate successfully with Istio injection enabled in the namespace.
+- **Should** avoid workarounds such as disabling strict mTLS peer authentication.
+
+### Network Policies
+
+- **Must** define network policies under the `allow` key as required.
+- **Should** minimize network policies to specific selectors needed for Ingress/Egress traffic.
+- **May** template network policy keys to provide flexibility for delivery customers to configure.
+
+### Keycloak
+
+- **Must** use and create a Keycloak client through the `sso` key if the application provides a user login.
```

Comment:
Should we also give some guidance about the client id and the secret name? More of a Developer/Admin experience thing. I'd say 'uds-core-<app>' for the client id and '<app>-sso' in the <app> namespace for the secret is the common pattern right now.

Reply:
We could though we'd need to make it work for leapfrog too (which doesn't use the `uds` brand).  IMO `uds-core-mattermost` is also a misnomer since mattermost is not in uds-core.  should be `uds-swf-mattermost` most likely.

Reply:
Chiming in my 2cents on the client id:
- I think the `uds-` prefix is fine, this is all in the context of things "made for uds" so I think we're okay enforcing that if we want
- It should definitely have the "app name/identifier" (i.e. mattermost, grafana, sonarqube, ...)
- Where applicable it may make sense to add the "app group" (i.e. core/swf/...)

I don't know if it makes sense to be prescriptive on the secret name - for reference the `neuvector` chart only allows you to mount secrets that match a specific name (`neuvector-init`) and does not expose values to point at a different secret. This is probably not common but I would hesitate to strictly enforce this one.